### PR TITLE
Quickpay v10 fixes

### PR DESCRIPTION
**Pull request by ta**
This PR makes all Payments/Operations requests to QuickPay v10 platform synchronous ensuring that the payment's status is in sync between AM and QuickPay

### CODE REVIEW

File: test/unit/gateways/quickpay_v10_test.rb [hunk 1]
```diff
@@ -32,9 +32,11 @@ def test_successful_purchase
       parsed = parse(data)
       if parsed['order_id']
         assert_match %r{/payments}, endpoint
-      else
+      elsif !parsed['auto_capture'].nil?
         assert_match %r{/payments/\d+/authorize}, endpoint
-        assert_equal parsed['auto_capture'], true 
+        assert_equal parsed['auto_capture'], false
```
Comment (aprofeit) [inline test/unit/gateways/quickpay_v10_test.rb]: Very minor nitpick, but these should be flipped. `assert_equal expected, actual`

File: test/remote/gateways/remote_quickpay_v10_test.rb [hunk 2]
```diff
@@ -128,15 +140,26 @@ def test_successful_purchase_and_credit
...
+    assert refund = @gateway.refund(@amount, auth.authorization)
+    assert_failure refund
+    assert_equal 'Rejected test operation', refund.message
+  end
+  
   def test_successful_store
-    assert response = @gateway.store(@valid_card, @options.merge(:description => 'test'))
-    assert_success response
+    assert response = @gateway.store(@valid_card, @options.merge(:description => 'test', :currency => 'USD', :amount => @amount))
+    assert_success response
```
Comment (aprofeit) [inline test/remote/gateways/remote_quickpay_v10_test.rb]: Another minor thing, have some extra whitespace here as well as lines 161 and 162.

File: lib/active_merchant/billing/gateways/quickpay/quickpay_v10.rb [hunk 3]
```diff
@@ -182,13 +197,24 @@ def parse(body)
...
 
           !(has_error || invalid_code)
         end
 
         def message_from(success, response)
-          success ? 'OK' : (response['message'] || response['qp_status_msg'])
+          success ? 'OK' : (response['message'] || invalid_operation_message(response))
+        end
+        
+        def invalid_operation_code response
```
Comment (aprofeit) [inline lib/active_merchant/billing/gateways/quickpay/quickpay_v10.rb]: Although ruby allows you to omit braces on method definitions, they should still be there. Please add brackets `def invalid_operation_code(response)`. Same thing for `def invalid_operation_message(response)`.

**Comment by ta** [comment]
@aprofeit Thanks a lot for reviewing so promptly. We've now incorporated the changes you've asked.

**Pull request closed without merging**